api.py

```diff
@@ -1,3 +1,118 @@
+                            item["room"])
+            if group in change.group:
+                changes.append(change.get_str())
+        if changes:
+            return '\n'.join(changes)
+        return None
+
+    # --------------------------------------------------------------------------------------------------------------------
+    #                                         Consultations
+    # --------------------------------------------------------------------------------------------------------------------
+
+    def get_consultations(self):
+        r = requests.get(self.api_url + 'consultations')
+        consultations_json = eval(json.dumps(r.json()).replace("null", "None"))["data"]
+        consultations = []
+        for item in consultations_json:
+            for temporal in item["times"]:
+                consultation = Consultation(item["teacher"], item["room"], item["email"], item["department"],
+                                            item["times"], temporal["weekday"], temporal["time"])
+            consultations.append(consultation.get_str())
+        if consultations:
+            return '\n'.join(consultations)
+        return None
+
+    def get_consultations_by_teacher(self, teacher):
+        r = requests.get(self.api_url + 'consultations')
+        consultations_json = eval(json.dumps(r.json()).replace("null", "None"))["data"]
+        consultations = []
+        for item in consultations_json:
+            for temporal in item["times"]:
+                consultation = Consultation(item["teacher"], item["room"], item["email"], item["department"],
+                                            item["times"], temporal["weekday"], temporal["time"])
+            if teacher in consultation.teacher:
+                consultations.append(consultation.get_str())
+        if consultations:
+            return '\n'.join(consultations)
+        return None
+
+    def get_consultations_by_department(self, department):
+        r = requests.get(self.api_url + 'consultations?department='+department)
+        consultations_json = eval(json.dumps(r.json()).replace("null", "None"))["data"]
+        consultations = []
+        for item in consultations_json:
+            for temporal in item["times"]:
+                consultation = Consultation(item["teacher"], item["room"], item["email"], item["department"],
+                                            item["times"], temporal["weekday"], temporal["time"])
+            consultations.append(consultation.get_str())
+        if consultations:
+            return '\n'.join(consultations)
+        return None
+
+    # --------------------------------------------------------------------------------------------------------------------
+    #                                         Teacher searching
+    # --------------------------------------------------------------------------------------------------------------------
+
+    def search_teachers(self):
```

Comment:
Why teacher searching should be here? And what info that should contain, if rooms and timetable always changes?